src/databases/Mili/avtMiliMetaData.C

```diff
@@ -0,0 +1,3126 @@
+
+
+// ***************************************************************************
+//  Method: avtMiliMetaData::SetNumClasses
+//
+//  Purpose:
+//      Set the number of classes in our dataset, and initialize the 
+//      miliClasses container. 
+//
+//  Arguments: 
+//      nClasses    The number of classes in our dataset.
+//           
+//  Programmer: Alister Maguire
+//  Creation:   Jan 15, 2019
+//
+//  Modifications:
+//
+// ****************************************************************************
+
+void
+avtMiliMetaData::SetNumClasses(int nClasses)
+{
+    if (miliClasses != NULL)
+    {
+        for (int i = 0; i < numClasses; ++i)
+        {
+            if (miliClasses[i] != NULL)
+            {
+                delete miliClasses[i];
+            }
+        }
+
+        delete [] miliClasses;
+    }
+
+    numClasses  = nClasses;
+    miliClasses = new MiliClassMetaData *[numClasses];
+    for (int i = 0; i < numClasses; ++i)
+    {
+        miliClasses[i] = NULL;
+    }
+}
+
+
+// ***************************************************************************
+//  Method: avtMiliMetaData::AddClassMD
+//
+//  Purpose:
+//      Add a mili Class meta data object to our container. 
+//
+//  Arguments: 
+//      classIdx    The index of the Clas md. 
+//      mcmd        The mili class meta data. 
+//
+//      NOTE: mcmd enters ownership of avtMiliMetaData and all
+//            memory is managed internally from then on. 
+//           
+//  Programmer: Alister Maguire
+//  Creation:   Jan 15, 2019
+//
+//  Modifications:
+//
+// ****************************************************************************
+
+void
+avtMiliMetaData::AddClassMD(int classIdx, 
+                            MiliClassMetaData *mcmd)
+{
+    if (classIdx < 0 || classIdx >= numClasses)
+    {
+        char expected[1024]; 
+        char recieved[1024];
+        sprintf(expected, "an index betwen 0 and %d", numClasses - 1);
+        sprintf(recieved, "%d", classIdx);
+        EXCEPTION2(UnexpectedValueException, expected, recieved);
+    }
+
+    if (miliClasses == NULL)
+    {
+        char msg[1024];
+        sprintf(msg, "Attempting to add MD to uninitialized container!");
+        EXCEPTION1(ImproperUseException, msg);
+    }
+
+    if (miliClasses[classIdx] != NULL)
+    {
+        delete miliClasses[classIdx];
+    }
+
+    miliClasses[classIdx] = mcmd;
+}
+
+
+// ***************************************************************************
+//  Method: avtMiliMetaData::SetNumCells
+//
+//  Purpose:
+//      Set the number of cells for a given domain. 
+//
+//  Arguments: 
+//      domain    The domain of interest. 
+//      nCells    The number of cells on the given domain. 
+//
+//  Programmer: Alister Maguire
+//  Creation:   Jan 15, 2019
+//
+//  Modifications:
+//
+// ****************************************************************************
+
+void
+avtMiliMetaData::SetNumCells(int domain, int nCells)
+{
+    if (domain >= 0 and domain < numDomains)
```

Comment:
MS compiler error for 'and', should be '&&'.

Reply:
Wow, I didn't even realize "and" was allowed in c++. That was just my brain thinking in python. Just fixed.

Reply:
Thanks.